linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseClassifiers.scala

```diff
@@ -58,14 +60,32 @@ object ResponseClassifiers {
     }
   }
 
+  object RetryableException {
+    def unapply(e: Throwable): Boolean = e match {
+      case RetryableWriteException(_) => true
+      case Failure(Some(_: TimeoutException)) => true
+      case Failure(Some(_: UtilTimeoutException)) => true
+      case _: TimeoutException => true
+      case _: UtilTimeoutException => true
+      case _: ChannelClosedException => true
+      case _ => false
+    }
```

Comment:
This is almost identical to Finagle's `TimeoutAndWriteExceptionsOnly` partial function, defined here:

https://github.com/twitter/finagle/blob/develop/finagle-core/src/main/scala/com/twitter/finagle/service/RetryPolicy.scala#L162

That function does not catch ChannelClosedException.  But maybe you could do something like:

    TimeoutAndWriteExceptionsOnly.orElse(ChannelClosedExceptionsOnly)

I think tying ourselves to their definition of retryable exceptions is worthwhile, since it may change over time.